py-polars/tests/unit/series/test_series.py

```diff
@@ -2340,5 +2340,42 @@ def test_search_sorted(
 
 
 def test_series_from_pandas_with_dtype() -> None:
-    s = pl.Series("foo", pd.Series([1, 2, 3]), pl.Float32)
-    assert_series_equal(s, pl.Series("foo", [1, 2, 3], dtype=pl.Float32))
+    expected = pl.Series("foo", [1, 2, 3], dtype=pl.Int8)
+    s = pl.Series("foo", pd.Series([1, 2, 3]), pl.Int8)
+    assert_series_equal(s, expected)
+    s = pl.Series("foo", pd.Series([1, 2, 3], dtype="Int16"), pl.Int8)
+    assert_series_equal(s, expected)
+
+    with pytest.raises(pl.ComputeError, match="conversion from"):
+        pl.Series("foo", pd.Series([-1, 2, 3]), pl.UInt8)
+    s = pl.Series("foo", pd.Series([-1, 2, 3]), pl.UInt8, strict=False)
+    assert s.to_list() == [None, 2, 3]
+    assert s.dtype == pl.UInt8
+
+    with pytest.raises(pl.ComputeError, match="conversion from"):
+        pl.Series("foo", pd.Series([-1, 2, 3], dtype="Int8"), pl.UInt8)
+    s = pl.Series("foo", pd.Series([-1, 2, 3], dtype="Int8"), pl.UInt8, strict=False)
+    assert s.to_list() == [None, 2, 3]
+    assert s.dtype == pl.UInt8
+
+
+def test_series_from_pyarrow_with_dtype() -> None:
+    s = pl.Series("foo", pa.array([-1, 2, 3]), pl.Int8)
+    assert_series_equal(s, pl.Series("foo", [-1, 2, 3], dtype=pl.Int8))
+
+    with pytest.raises(pl.ComputeError, match="conversion from"):
+        pl.Series("foo", pa.array([-1, 2, 3]), pl.UInt8)
+
+    s = pl.Series("foo", pa.array([-1, 2, 3]), dtype=pl.UInt8, strict=False)
+    assert s.to_list() == [None, 2, 3]
+    assert s.dtype == pl.UInt8
+
+
+def test_series_from_pandas_with_strict() -> None:
+    s = pl.Series(pd.Series([1, 2.5, 3]), strict=False)
+    assert_series_equal(s, pl.Series([1.0, 2.5, 3.0], dtype=pl.Float64))
+
+
+def test_series_from_pyarrow_with_strict() -> None:
+    s = pl.Series(pa.array([1, 2.5, 3]), strict=False)
+    assert_series_equal(s, pl.Series([1.0, 2.5, 3.0], dtype=pl.Float64))
```

Comment:
I might be missing something, but I don't see what these tests add to what you already have

Rather than adding these, how about testing converting from a numpy array with `[-1, 2, 3]` to a Polars Series with dtype `pl.UInt8`, with strict=False and strict=True? I think this will cover the behaviour change you mentioned in https://github.com/pola-rs/polars/pull/15962/files#r1590912969

Reply:
The two tests added earlier were because I felt there needed to be a simple test with `strict=False` without any other parameters. However, I think you are right, these two tests are covered by others.

The behavior change mentioned in https://github.com/pola-rs/polars/pull/15962#discussion_r1590912969 has been covered by the test using `strict=False` like below. It was precisely because of your previous suggestion to modify the test that I discovered this potential problem.

```python
s = pl.Series("foo", pd.Series([-1, 2, 3], dtype="Int8"), pl.UInt8, strict=False)
assert s.to_list() == [None, 2, 3]
assert s.dtype == pl.UInt8
```

Nevertheless, I will add a test with a numpy array and the `strict` parameter.